### DISCUSSION TITLE
Is this still active? What still needs to be done?

### DISCUSSION THREAD

**Discussion by CyrusAlbright** [rust-wii/luma#13] [General]
Hey, I'm interested in developing for the Wii using Rust. Is this something that's being worked on still and what needs to be done to make this feature complete/on par with libogc?

I'd be happy to help, but I don't know what I would start with because there are no active issues here. I'm familiar enough with C and I can bumble my way through assembly as well.

Thanks in advance, and I look forward to potentially working on this project!

**Comment by FrictionlessPortals** [comment]
Hi! 

This project is still being developed! Unfortunately due to studies, I lost the time work on new features for it but I am still active to review / pull PRs and respond to issues, thankfully I am gaining more time to work on side-projects as of recently and would like to pick this back up.

Currently, this project requires a better debugging implementation, at the moment there is an open draft PR for USB Gecko support through Dolphin and potentially the Wii, but I am unsure as I don't own any device to check this on the Wii.

There is also a implementation by @linkmauve for the VI subsystem and an example to draw on the Wii using it, this could be expanded to have the fully featured GX module like in libogc, this module however is very large and would take some time to rewrite.

Before I lost time, I was working on porting the threaded system based on spinlock as implemented in libogc, in hope that giving the library a threaded system could allow porting of crates to allow minimal rewriting for networking and such.

If you have any further questions feel free to ask!

There is a small Discord for this project also: https://discord.gg/ABKmp4TPZA

**Comment by CyrusAlbright** [comment]
Oh nice, alright. I don't have a USB Gecko device either so I'm unsure of how much I'll be able to help there. But I would definitely be interested in working on the graphics side of things, or on multithreading. I'll join the Discord and we can work from there.
